### ISSUE TITLE
Timeout error in m.request doesn't seem to be handled

### ISSUE THREAD

**Issue by RadhiFadlillah** [MithrilJS/mithril.js#2559]
> This issue is follow up from my question in Gitter

**Mithril version:** 2.0.4

**Browser and OS:** Firefox 71.0 in Manjaro Linux 64-bit

## Steps to Reproduce

1. Create some XHR request and specify its timeout. You might want to throttle your connection speed.
2. Send the request.
3. Watch the error response.

In my case, the code looks like this

```javascript
m.request("/api/accounts", { timeout: 500 })
    .then(json => {
        state.tableContents = json
    })
    .catch(err => {
        console.log(err.code, err.message)
        state.dlgError.message = err.message
        state.dlgError.visible = true
    })
    .finally(() => {
        state.isLoading = false
    })
```

## Expected Behavior

It should give me an obvious error message. In my case, the console should print something like :

```
request timeout after 500ms
```

## Current Behavior

The timeout error doesn't seem to be handled properly. From my code above, the console print error report like this :

```
0 null
```

**Comment by isiahmeadows** [comment]
Note for later: there's a `timeout` event I can use to detect timeouts independently of other network errors.

**Comment by barneycarroll** [comment]
Fixed in #2646
